videosink/encoder.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package videosink
+
+import (
+	"image/jpeg"
+	"image/png"
+	"sync"
+)
+
+var jpegOptions = jpeg.Options{
+	Quality: 95,
+}
+
+type pngEncoderBufferPool sync.Pool
+
+var _ png.EncoderBufferPool = (*pngEncoderBufferPool)(nil)
+
+func (p *pngEncoderBufferPool) Get() *png.EncoderBuffer {
+	if buf := (*sync.Pool)(p).Get(); buf != nil {
```

Comment:
Can you confirm if this works?

```
buf, _ := (*sync.Pool)(p).Get().(*png.EncoderBuffer)
return buf
```

Reply:
Yes, it does and I changed the code. During development I originally wrote `return (*sync.Pool)(p).Get().(*png.EncoderBuffer)` at first, but that triggers a panic at runtime: `interface conversion: interface {} is nil, not *png.EncoderBuffer`.